adobe/react-spectrum
# fix(#2449): ActionBar: Restore focus to SelectableCollection on Esc

Closes #2449 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #2449. 

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/973b3aaf0fbb9bd7407c220817dc5b899e8f7952/storybook/index.html?path=/story/actionbar--default
2. Keyboard navigate to an item, or the checkbox for the item, in the TableView.
3. Press Space key to select the item within the TableView, which should expand the ActionBar.
4. Press Tab key to navigate to first action button within the ActionBar toolbar.
5. Press ArrowRight key to move focus to the "More Actions" menu button.
6. Press Space key to expand the "More Actions" menu.
7. Press Esc key to close the "More Actions" menu.
8. Focus should restore to the "More Actions" menu button without clearing the TableView selection.
9. Press Esc key again to clear selections and dismiss the ActionBar.
10. Focus should restore to the item, or the checkbox or cell within the item, in the TableView where focus left off.

It may be useful testing the following behavior using a tool such as [NerdeFocus](https://chrome.google.com/webstore/detail/nerdefocus/lpfiljldhgjecfepfljnbjnbjfhennpd?hl=en-US) to log items as they receive focus.

12. Press Space key to select the item within the TableView again, which should expand the ActionBar.
13. Scroll the selected item out of view within the TableView body.
14. Because the collection is virtualized, the focus will shift the the TableView itself.
15. Press Tab key to navigate to first action button within the ActionBar toolbar.
16. Press ArrowRight key to move focus to the "More Actions" menu button.
17. Press Space key to expand the "More Actions" menu.
18. Press Esc key to close the "More Actions" menu.
19. Focus should restore to the "More Actions" menu button without clearing the TableView selection.
20. Press Esc key again to clear selections and dismiss the ActionBar.
21. Focus should restore to the item, or the checkbox or cell within the item, in the TableView where focus left off, but in order to do so, the TableView will receive focus first, the scroll the appropriate virtualized item into view and then move focus to it.

## 🧢 Your Project:

Adobe/Accessibility
